### ISSUE TITLE
How to analyze subsets of the simulation

### ISSUE THREAD

**Issue by elkhoury** [choderalab/yank#792]
We would like to analyze a subset of the simulation (e.g. 1000 iterations out of 20000 iterations) in order to have the calculated free energies and the convergence plots.

Is there a way to automatically analyze subsets of the data?

Thank you,

Léa El Khoury

(@dmobley, @renm1)

**Comment by jchodera** [comment]
This sounds like something we should be able to easily add to the CLI for analysis. We already support `--start XXX` and `--end YYY` for [trajectory analysis](https://github.com/choderalab/yank/blob/master/Yank/commands/analyze.py#L55-L56), so we'd just have to pass these on to [`yank.analysis.analyze_directory()`](https://github.com/choderalab/yank/blob/master/Yank/commands/analyze.py#L81) and modify [this code](https://github.com/choderalab/yank/blob/master/Yank/analyze.py#L1361-L1387) to handle that.

**Comment by Lnaden** [comment]
Should be easy enough. There is no public API way to do this right now, but you can follow the logic of the `analyze_directory` function in the source code @jchodera linked to piece together a temporary solution. 

@elkhoury by "convergence plots" do you mean in the Jupyter Notebooks plots from `yank analyze report`?

**Comment by elkhoury** [comment]
Yes, I mean the convergence plots generated in the Jupyter Notebooks. 

Thank you @jchodera, @lnaden

**Comment by andrrizzi** [comment]
The `Analyzer` classes now have a `Analyzer.max_n_iterations` property that you can use to limit the number of iterations you want to analyze (implemented in #915). There's still no hook through the CLI for this, but for now, you can add
```python
analyzer_kwargs['max_n_iterations'] = 1000
```
in the first cell of the Jupyter notebook.
